### PR TITLE
[MAINT] fix pendingdeprecationwarning

### PR DESCRIPTION
**Pull request by alexrockhill**
Fixes issue related to https://github.com/mne-tools/mne-python/pull/10825 as in https://github.com/mcmtroffaes/sphinxcontrib-bibtex/pull/302.

### CODE REVIEW

**Comment by alexrockhill** [comment]
Okay to merge? Thanks for a great extension 😄

File: sphinx_design/dropdown.py [hunk 1]
```diff
@@ -142,7 +142,11 @@ class DropdownHtmlTransform(SphinxPostTransform):
 
     def run(self):
         """Run the transform"""
-        for node in self.document.traverse(lambda node: is_component(node, "dropdown")):
+        # Can just use "findall" once docutils 0.18+ is required
```
Comment (drammock) [inline sphinx_design/dropdown.py:145]: the change was introduced in 0.18.1
```suggestion
        # Can just use "findall" once docutils 0.18.1+ is required
```

File: sphinx_design/tabs.py [hunk 2]
```diff
@@ -213,7 +213,9 @@ def get_unique_key(self):
 
     def run(self) -> None:
         """Run the transform."""
-        for tab_set in self.document.traverse(
+        # Can just use "findall" once docutils 0.18+ is required
```
Comment (drammock) [inline sphinx_design/tabs.py:216]:
```suggestion
        # Can just use "findall" once docutils 0.18.1+ is required
```

File: sphinx_design/dropdown.py [hunk 3]
```diff
@@ -142,7 +142,11 @@ class DropdownHtmlTransform(SphinxPostTransform):
 
     def run(self):
         """Run the transform"""
-        for node in self.document.traverse(lambda node: is_component(node, "dropdown")):
+        # Can just use "findall" once docutils 0.18.1+ is required
+        meth = "findall" if hasattr(self.document, "findall") else "traverse"
+        for node in getattr(self.document, meth)(
```
Comment (chrisjsewell) [inline sphinx_design/dropdown.py:147]:
```suggestion
        for node in getattr(self.document, "findall", self.document.traverse)(
```

this is more sucinct no?

**Comment by chrisjsewell** [comment]
Thanks, the only reason I didn't #74 though, was that I had a slight worry that, since `findall` is a generator, rather than a pre-created list, there might be an issue with `node.replace_self`, since it is altering the thing you are generating from.
I wanted to double-check that this would not end up with any unexpected behaviour

File: sphinx_design/dropdown.py (hunk 3 above)
Comment (larsoner) [reply sphinx_design/dropdown.py]: If they do actually remove `.traverse` this new version breaks, hence the `hasattr` approach. But maybe by then you will  have moved to requiring a new enough version anyway...

**Comment by alexrockhill** [comment]
Are we still worried about unexpected behavior or is this okay to merge?

File: sphinx_design/dropdown.py [hunk 4]
```diff
@@ -142,7 +142,10 @@ class DropdownHtmlTransform(SphinxPostTransform):
 
     def run(self):
         """Run the transform"""
-        for node in self.document.traverse(lambda node: is_component(node, "dropdown")):
+        # Can just use "findall" once docutils 0.18.1+ is required
+        for node in getattr(self.document, "findall", self.document.traverse)(
+            lambda node: is_component(node, "dropdown")
+        ):
```
Comment (drammock) [inline sphinx_design/dropdown.py:148]:
```suggestion
        # use this instead once docutils>=0.18.1 is min supported version:
        # _dropdowns = list(self.document.findall(lambda node: is_component(node, "dropdown")))
        _meth = getattr(self.document, "findall", self.document.traverse)
        _dropdowns = list(_meth(lambda node: is_component(node, "dropdown")))
        for node in _dropdowns:
```

File: sphinx_design/tabs.py [hunk 5]
```diff
@@ -213,7 +213,9 @@ def get_unique_key(self):
 
     def run(self) -> None:
         """Run the transform."""
-        for tab_set in self.document.traverse(
+        # Can just use "findall" once docutils 0.18.1+ is required
+        meth = "findall" if hasattr(self.document, "findall") else "traverse"
+        for tab_set in getattr(self.document, meth)(
             lambda node: is_component(node, "tab-set")
         ):
```
Comment (drammock) [inline sphinx_design/tabs.py:220]:
```suggestion
        # use this instead once docutils>=0.18.1 is min supported version:
        # _tab_sets = list(self.document.findall(lambda node: is_component(node, "tab-set")))
        _meth = getattr(self.document, "findall", self.document.traverse)
        _tab_sets = list(_meth(lambda node: is_component(node, "tab-set")))
        for tab_set in _tab_sets:
```

**Review by drammock — COMMENTED** [review]
Bump.  Our CIs are still failing because of this.  Any time to review @chrisjsewell?  I've made suggestions below that cast the generators to lists before the for-loop, so it should deal with your worry about `node.replace_self` (I'm not sure whether or not it's actually a problem, but this way it doesn't matter).

**Comment by drammock** [comment]
Bump again. I think my suggested change addresses the only worry you've so far expressed @chrisjsewell.  Can you find time to review?

**Comment by chrisjsewell** [comment]
Heya, thanks for the PR, but I've superseded it with #94, to use the already available "compat" function 😄

**Pull request closed without merging**